### PR TITLE
Add ecosystem-specific code owners

### PR DESCRIPTION
**Pull request by mattt**
This PR updates the [CODEOWNERS file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to have ecosystem-specific teams suggested as reviewers on relevant PRs. For example, the team of Java reviewers would be suggested for a PR that touches either the `gradle` or `maven` subdirectories.

Before merging this PR, we need to create the corresponding teams, likely as sub-teams to @dependabot/reviewers. After merging this PR, we should update our branch protection rules to require approval by the codeowners.

### CODE REVIEW

**Comment by jurre** [comment]
Nice one, I think this makes sense 👍 I'm not entirely sure that we have folks to staff all these teams, so that's something that we'll need to figure out.

**Comment by mattt** [comment]
Before merging, we need to create the listed teams (e.g. `@dependabot/ruby-reviewers`) in our GitHub organization.

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1,17 @@
...
+
+# Ecosystems
+bundler/* @dependabot/ruby-reviewers
+cargo/* @dependabot/rust-reviewers
+composer/* @dependabot/php-reviewers
+docker/* @dependabot/docker-reviewers
+elm/* @dependabot/elm-reviewers
+go_modules/* @dependabot/go-reviewers
+gradle/* @dependabot/java-reviewers
+hex/* @dependabot/erlang-reviewers
```
Comment (jurre) [inline CODEOWNERS:11]:
```suggestion
hex/* @dependabot/elixir-reviewers
```

File: CODEOWNERS [hunk 2]
```diff
@@ -1 +1,17 @@
 * @dependabot/reviewers
```
Comment (mattt) [inline CODEOWNERS:1]:
```suggestion
* @dependabot/maintainers
```

**Pull request merged**